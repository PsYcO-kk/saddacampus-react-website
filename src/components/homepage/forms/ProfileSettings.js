import React from "react";

export default class ProfileSettings extends React.Component {
  
  render(){
        return(
            <form>
				<div id="profileSettings">
					<div className="form-group">
						<label htmlFor="avatar">Upload an Avatar:</label>
						<input type="file" className="form-control-file" id="avatar" />
					</div>
					<div className="form-group">
						<input type="text" className="form-control" id="username" placeholder="Enter Username" minLength="5" maxLength="25" required />
						<div id="feedback"></div>
					</div>
					<button id="create" className="btn btn-success" type="submit">Create Account</button>
				</div>
				<br />
			</form>
        );
    }
}
