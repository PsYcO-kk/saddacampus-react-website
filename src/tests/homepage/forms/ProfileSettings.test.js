import React from 'react';
import { shallow } from 'enzyme';
import ProfileSettings from '../../../components/homepage/forms/ProfileSettings';

let wrapper, country_code='+91', number='5464645454';

beforeEach(() => {
	wrapper = shallow(<ProfileSettings country_code={country_code} number={number} />);
});

test('should render ProfileSettings correctly', () => {
	expect(wrapper).toMatchSnapshot();
});

// test('should set and render invalid-feedback when (username.length < 5)', () => {
// 	const value = '@5ec';
// 	wrapper. find('#username').simulate('change', {
// 		target: { value }
// 	});
// 	expect(wrapper.state('feedback')).toBe('Username must be within 5-25 characters.');
// 	expect(wrapper).toMatchSnapshot();
// });

// test('should set and render invalid-feedback when (username.length > 25)', () => {
// 	const value = 'rjec y4 389ty94h eiuw390$R%$#@gjjg';
// 	wrapper. find('#username').simulate('change', {
// 		target: { value }
// 	});
// 	expect(wrapper.state('feedback')).toBe('Username must be within 5-25 characters.');
// 	expect(wrapper).toMatchSnapshot();
// });
