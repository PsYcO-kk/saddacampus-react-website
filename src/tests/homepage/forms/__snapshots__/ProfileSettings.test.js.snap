// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProfileSettings correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="avatar"
    >
      Upload an Avatar:
    </label>
    <input
      className="form-control-file"
      id="avatar"
      type="file"
    />
  </div>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      id="username"
      onChange={[Function]}
      placeholder="Enter Username"
      required={true}
      type="text"
      value=""
    />
    <div
      id="feedback"
    />
  </div>
  <button
    className="btn btn-success"
    disabled={true}
    type="submit"
  >
    Create Account
  </button>
</form>
`;
